src/MOIWrapper.jl

```diff
@@ -1,408 +1,435 @@
-    copyto!(x, primal_row_solution(instance.inner))
-end
+# ===============================
+#   Optimize and post-optimize
+# ===============================
 
-function LQOI.get_variable_dual_solution!(instance::Optimizer, x::Vector{Float64})
-    copyto!(x, dual_column_solution(instance.inner))
+function MOI.optimize!(model::Optimizer)
+    Clp.initial_solve(model.inner)
+    model.optimize_called = true
+    return nothing
 end
 
-function LQOI.get_linear_dual_solution!(instance::Optimizer, x::Vector{Float64})
-    copyto!(x, dual_row_solution(instance.inner))
-end
+MOI.get(model::Optimizer, ::MOI.NumberOfVariables) = Clp.get_num_cols(model.inner)
 
-function LQOI.get_objective_value(instance::Optimizer)
-    return objective_value(instance.inner)
-end
+MOI.get(model::Optimizer, ::MOI.ObjectiveValue) = Clp.get_obj_value(model.inner)
 
-function LQOI.get_farkas_dual!(instance::Optimizer, result::Vector{Float64})
-    copyto!(result, infeasibility_ray(instance.inner))
-    LinearAlgebra.rmul!(result, -1.0)
-end
+function MOI.get(model::Optimizer, ::MOI.TerminationStatus)
 
-function LQOI.get_unbounded_ray!(instance::Optimizer, result::Vector{Float64})
-    copyto!(result, unbounded_ray(instance.inner))
-end
+    model.optimize_called || return MOI.OPTIMIZE_NOT_CALLED
+
+    st = Clp.status(model.inner)
 
-function LQOI.get_termination_status(instance::Optimizer)
-    status = ClpCInterface.status(instance.inner)
-    if status == 0
+    if st == 0
         return MOI.OPTIMAL
-    elseif status == 1
+    elseif st == 1
         return MOI.INFEASIBLE
-    elseif status == 2
+    elseif st == 2
         return MOI.DUAL_INFEASIBLE
-    elseif status == 3
-        return MOI.OTHER_LIMIT
-    elseif status == 4
-        return MOI.OTHER_ERROR
+    elseif st == 3
+        # No more granular information that "some limit is reached"
+        return MOI.OTHER_LIMIT  
     else
-        error("status returned was $(status), but it must be in [0,1,2,3,4]")
+        return MOI.OTHER_ERROR
     end
 end
 
-function LQOI.get_primal_status(instance::Optimizer)
-    if is_proven_dual_infeasible(instance.inner)
-        return MOI.INFEASIBILITY_CERTIFICATE
-    elseif primal_feasible(instance.inner)
+function MOI.get(model::Optimizer, ::MOI.ResultCount)
+    st = MOI.get(model, MOI.TerminationStatus())
+
+    # TODO: check that this is correct
+    if st == MOI.OPTIMIZE_NOT_CALLED || st == MOI.OTHER_ERROR
+        return 0
+    end
+    return 1
+end
+
+function MOI.get(model::Optimizer, ::MOI.PrimalStatus)
+    if Clp.is_proven_dual_infeasible(model.inner)
+        return MOI.UNBOUNDED_RAY
+    elseif Clp.primal_feasible(model.inner)
         return MOI.FEASIBLE_POINT
     else
         return MOI.UNKNOWN_RESULT_STATUS
     end
 end
 
-function LQOI.get_dual_status(instance::Optimizer)
-    if is_proven_primal_infeasible(instance.inner)
+function MOI.get(model::Optimizer, ::MOI.DualStatus)
+    if Clp.is_proven_primal_infeasible(model.inner)
```

Comment:
Call `MOI.check_result_index_bounds(model, attr)`

Reply:
What does `MOI.check_result_index_bounds(model, attr)` do exactly? Is it to check that a result does exist before returning one?

Reply:
Yes, it checks with the value of `ResultCount`

Reply:
FYI, `MOI.check_result_index_bounds` will throw an error when querying `DualStatus(2)`, while the test [`solve_result_index`](https://github.com/JuliaOpt/MathOptInterface.jl/blob/ee77d85c69b8d32b8455376b40b0698e52312341/src/Test/UnitTests/solve.jl#L141) expects an `MOI.NO_SOLUTION`.

I've implemented things so as to pass tests in `solve_result_index`; I did find it a bit weird that `MOI.get(model, MOI.PrimalStatus(2))` is expected to not return an error, but  `MOI.get(model, MOI.VariablePrimal(2), x)` is expected to. Won't prevent me from sleeping though.

Reply:
See https://github.com/JuliaOpt/MathOptInterface.jl/pull/912 for the relevant discussion